File: Lab_5/OP_lab5.py
import math
def prime_check(num):
    divisor = 2
    while num % divisor != 0 and divisor**2 <= num:
        divisor += 1
    return divisor**2 > num
        
    
n = int(input("Enter n: "))
for i in range(2,int(math.log2(n+1))+1):
    if(prime_check(i)):
        mers_numb = 2**i - 1
        if(prime_check(mers_numb)):
            print(mers_numb)

File: Lab_5/OP_lab_5/OP_lab_5/OP_lab_5.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main()
{
	long long n;
	cout << "Enter n: ";
	cin >> n;
	for (long long i = 2; i <= log2(n + 1); i++) {
		long long divisor=2;
		while (i % divisor != 0 && divisor <= sqrt(i)) {
			divisor++;
		}
		if (divisor>sqrt(i)) {
			long long mers_numb = pow(2, i) - 1;
			divisor = 2;
			while (mers_numb % divisor != 0 && divisor <= sqrt(mers_numb)) {
				divisor++;
			}if (divisor > sqrt(mers_numb)) cout << mers_numb<<" ";
		}
	}
}